okta/data_source_okta_app_saml.go

```diff
@@ -319,11 +319,22 @@ func dataSourceAppSamlRead(ctx context.Context, d *schema.ResourceData, m interf
 		if len(appList) < 1 {
 			return diag.Errorf("no SAML application found with provided filter: %s", filters)
 		}
-		if filters.Label != "" && appList[0].Label != filters.Label {
-			return diag.Errorf("no SAML application found with the provided label: %s", filters.Label)
+
+		if filters.Label != "" {
+			foundMatch := false
+			for _, appItx := range appList {
+				if appItx.Label == filters.Label {
+					app = appItx
+					foundMatch = true
```

Comment:
Should it break here, similar to https://github.com/okta/terraform-provider-okta/blob/master/okta/data_source_okta_app.go#L111?